static/components.js

```diff
@@ -73,6 +73,13 @@ define(function (require) {
                 componentName: 'diff',
                 componentState: {}
             };
+        },
+        getOptViewWith: function (id, source, optimization) {
```

Comment:
Thanks for working out how this all fits together. It's a bit of a confusing mess!